src/Analyzers/CSharp/CodeFixes/FixIncorrectConstraint/CSharpFixIncorrectConstraintCodeFixProvider.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.FixIncorrectConstraint
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.FixIncorrectConstraint), Shared]
+    internal class CSharpFixIncorrectConstraintCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        private const string CS9010 = nameof(CS9010); // Keyword 'enum' cannot be used as a constraint.Did you mean 'struct, System.Enum'?	Net6 C:\github\repo_find_refs\Net6\Class1.cs 1	Active
+        private const string CS9011 = nameof(CS9011); // 'delegate' cannot be used as a constraint.Did you mean 'System.Delegate'?	Net6 C:\github\repo_find_refs\Net6\Class1.cs 1	Active
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpFixIncorrectConstraintCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(CS9010, CS9011);
+
+        internal override CodeFixCategory CodeFixCategory
+            => CodeFixCategory.Compile;
+
+        private static bool TryGetConstraint(
+            Diagnostic diagnostic,
+            CancellationToken cancellationToken,
+            [NotNullWhen(true)] out TypeConstraintSyntax? constraint,
+            out SyntaxToken enumOrDelegateKeyword)
+        {
+            enumOrDelegateKeyword = default;
+            constraint = diagnostic.Location.FindNode(cancellationToken) as TypeConstraintSyntax;
+            if (constraint == null)
+                return false;
```

Comment:
Just to confirm, most of the cases returning `false` seem unreachable. This is just being defensive, right?

Reply:
Correct.  Compiler diagnostics come with no contracts/guarantees. They can and do change over time. So consumption code needs to be written to be resilient to that.